app/src/main/java/com/karumi/ui/view/MainActivity.kt

```diff
@@ -1,28 +1,60 @@
 package com.karumi.ui.view
 
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
 import android.view.View
 import com.github.salomonbrys.kodein.Kodein.Module
 import com.github.salomonbrys.kodein.bind
 import com.github.salomonbrys.kodein.instance
 import com.github.salomonbrys.kodein.provider
 import com.karumi.R
 import com.karumi.ui.LifecycleSubscriber
+import com.karumi.ui.domain.model.SuperHero
+import com.karumi.ui.domain.usecase.GetSuperHeroes
 import com.karumi.ui.presenter.SuperHeroesPresenter
+import com.karumi.ui.view.adapter.SuperHeroesAdapter
 import kotlinx.android.synthetic.main.main_activity.progress_bar
+import kotlinx.android.synthetic.main.main_activity.recycler_view
 
 class MainActivity : BaseActivity(), SuperHeroesPresenter.View {
-
     private val presenter: SuperHeroesPresenter by injector.instance()
+    private lateinit var adapter: SuperHeroesAdapter
 
     override fun getLayoutId(): Int = R.layout.main_activity
     override fun obtainPresenter(): LifecycleSubscriber = presenter
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        initializeAdapter()
+        initializeRecyclerView()
+    }
+
+    private fun initializeAdapter() {
+        adapter = SuperHeroesAdapter(presenter)
+    }
+
+    private fun initializeRecyclerView() {
+        recycler_view.setLayoutManager(LinearLayoutManager(this))
+        recycler_view.setHasFixedSize(true)
+        recycler_view.setAdapter(adapter)
+    }
+
     override fun hideLoading() {
         progress_bar.visibility = View.GONE
     }
 
+    override fun showSuperHeroes(superHeroes: List<SuperHero>) {
+        adapter.addAll(superHeroes)
+        adapter.notifyDataSetChanged()
+    }
+
     override val activityModules = Module(allowSilentOverride = true) {
-        bind<SuperHeroesPresenter>() with provider { SuperHeroesPresenter(this@MainActivity) }
+        bind<SuperHeroesPresenter>() with provider {
+            SuperHeroesPresenter(this@MainActivity,
+                instance())
+        }
+
+        bind<GetSuperHeroes>() with provider { GetSuperHeroes() }
```

Comment:
Do you need to bind a class even when the constructor is empty?

Reply:
yes this is mandatory :( this is one of the things that I don't like from kodein.

Reply:
Then I'd not use this dependency injector. If we have to define a dependency every time we need it it's going to be a pain in the ass. Imagine you need to use this use case in two different places and this use case depends on a repository. Based on this library you'll need to write this code twice:

```kotlin
bind<GetSuperHeroes>() with provider { GetSuperHeroes(new SuperHeroesRepository()) }
```

When the dependency tree is simple, this is not an issue but in the future, this will be a pain in the ass. We could reduce the boilerplate by moving this declaration to a module configured just one.

Reply:
Yes, I have a general module, commons implementation must to be in general module. I am ok with you that it's not the best solution. I don't know why does not automatic.